reactioncommerce/api-migrations
# feat: legacy authorization migrations

Add migration for legacy-authorization roles.

This will migrate all "legacy" roles to their new "keto" counterparts.

## Testing
1. Create a few random Groups but DO NOT create `accounts-manager` or `system-manager` groups
1. set the permissions of the groups to any combination of the following:
```
[
    "account/invite",
    "core",
    "create-product",
    "createProduct",
    "dashboard",
    "discounts/apply",
    "media/create",
    "media/delete",
    "media/update",
    "orders",
    "order/fulfillment",
    "order/view",
    "product/admin",
    "product/archive",
    "product/clone",
    "product/create",
    "product/publish",
    "product/update",
    "reaction-accounts",
    "reaction-email",
    "reaction-templates",
    "shipping",
    "shops/create",
    "tags",
    "tags/admin",
    "tags/edit",
    "taxes/read",
    "taxes/write",
    "owner",
    "admin"
  ]
```
1. Run the migrations
1. see that the existing groups were updated to no longer have any of the above listed permissions, but instead their related new permission.
1. See that both `accounts-manager` and `system-manager` groups were created
